### DISCUSSION TITLE
[mono repo] CLI app entry convention?

### DISCUSSION THREAD

**Discussion by loynoir** [denoland/deno#15537] [Q&A]
What is the recommend entry for CLI app  in deno?

https://github.com/golang-standards/project-layout

---
`REPO/cmd/hello-world/main.ts` ?

Simple, but what about multi entry?

---
`REPO/cmd/hello-world/hello.ts` ?
`REPO/cmd/hello-world/world.ts` ?

---
`REPO/cmd/hello/main.ts` ?
`REPO/cmd/world/main.ts` ?

---
`REPO/pkg/hello-world/cmd/hello.ts` ?
`REPO/pkg/hello-world/cmd/world.ts` ?

---
`REPO/pkg/hello-world/cmd/hello/main.ts` ?
`REPO/pkg/hello-world/cmd/world/main.ts` ?

**Comment by laughedelic** [comment]
I think this page covers it: https://deno.land/manual@v1.24.3/tools/script_installer
You can name it whatever you want, users can override the executable name. 
I think the informal convention is to name it `cli.ts` for the main entry point.

Previous discussions: 
- https://github.com/denoland/deno/discussions/9057
- https://github.com/denoland/deno/issues/5004
